tokio/src/sync/tests/loom_watch.rs

```diff
@@ -34,3 +35,37 @@ fn smoke() {
         th.join().unwrap();
     })
 }
+
+#[test]
+fn wait_for_test() {
+    loom::model(move || {
+        let (tx, mut rx) = watch::channel(false);
+
+        let tx_arc = Arc::new(tx);
+        let tx1 = tx_arc.clone();
+        let tx2 = tx_arc.clone();
+
+        let th1 = thread::spawn(move || {
+            for _ in 0..3 {
+                tx1.send_modify(|x| {
+                    // here we avoid unsetting the value in case
+                    // the value is already set to true.
+                    if !*x {
+                        // force the loop to run.
+                        *x = false;
+                    }
+                });
```

Comment:
Here you're just doing "if false, then set to false". That's equivalent to this:
```suggestion
                tx1.send_modify(|_x| {});
```